### DISCUSSION TITLE
Adding BullModule.forRootAsync to CliModule makes it hang forever

### DISCUSSION THREAD

**Discussion by simplenotezy** [jmcdo29/nest-commander#945] [Q&A]
If I add:

```ts
BullModule.forRootAsync({
  useFactory: ConfigFactory.bullConfig,
  inject: [ConfigService],
}),
```

To my CliModule it will always run, so if we have cron jobs that uses a cli command it will keep reporting as 'Running'.

If I remove the BullModule the command will exit as usual.

**Comment by jmcdo29** [comment]
Interesting, I would expect the `static run` to end up calling `await app.close()` after the command runner service's `run` method finished:
https://github.com/jmcdo29/nest-commander/blob/7ed3febe2a5b77d542de40f5dcb045193a9f641e/packages/nest-commander/src/command.factory.ts#L16-L23

https://github.com/jmcdo29/nest-commander/blob/7ed3febe2a5b77d542de40f5dcb045193a9f641e/packages/nest-commander/src/command.factory.ts#L56-L62

**Reply by simplenotezy** [reply]
Hmm yeah.. Should I turn it into an issue, or what would you like?

**Reply by jmcdo29** [reply]
Can you make a reproduction when you do so I can test it locally?

**Reply by simplenotezy** [reply]
Sure thing, here you go:

https://github.com/simplenotezy/bullmq-and-commander

You can run `docker-compose up` to get a redis instance spinning and then you can run `node dist/cli basic` and you'll notice it hangs. Note: it only happens when after adding the `TestProcessor` to `AppModule`. If you remove the processor, it will work fine.

**Reply by jmcdo29** [reply]
Thanks for the reproduction. I'm pretty certain this is an issue with the `@nestjs/bull` library, as it doesn't implement an `OnModuleClose` to handle closing the connection to bull or redis when the application shuts down. `nest-commander` is properly closing the NestJS application, but if that connection pool is still open, then the Node event loop never resolves and the process is never finished

**Reply by simplenotezy** [reply]
I see, thanks @jmcdo29 - I have created an issue for bull here: https://github.com/nestjs/bull/issues/1705

**Reply by jmcdo29** [reply]
Interesting. So Nest does have those methods (hard to find, but they are there). Digging in a bit more, I do see that the Queue's `close` is indeed called, so now it's a question of why the delay. It looks like after half a minute the application is properly closed (see screenshot), so it's likely just redis being slow to kill the connection. Very annoying to say the least
![image](https://github.com/jmcdo29/nest-commander/assets/28268680/47ee69b5-2396-4a3d-b0f6-a8074c37b67f)

**Reply by simplenotezy** [reply]
Hmm, interesting. Well, in my case it hangs for hours. If I look at my CLI I see it hanging if I forgot to press CTRL+C even after hours.

Can you think of a workaround? Redis closes down fast in production (outside CLI) on graceful shutdowns etc, so it's indeed odd.

So far I am calling exit(0), but that causes a non-graceful-shutdown unfortunately.

**Reply by simplenotezy** [reply]
Created a ticket for it here: https://github.com/jmcdo29/nest-commander/issues/1042
